Microsoft.Toolkit.Uwp/Helpers/AppPinManager/AppPinManager.cs

```diff
@@ -0,0 +1,192 @@
+                            {
+                                pinResult = PinResult.PinAlreadyPresent;
+                            }
+                        }
+                    }
+                    else
+                    {
+                        pinResult = PinResult.PinNotAllowed;
+                    }
+                }
+                else
+                {
+                    pinResult = PinResult.UnsupportedDevice;
+                }
+            }
+
+            return pinResult;
+        }
+
+        /// <summary>
+        /// Pin Specific App in Windows StartMenu
+        /// </summary>
+        /// <param name="entry">AppListEntry</param>
+        /// <returns>PinResult</returns>
+        public static async Task<PinResult> PinSpecificAppToStartMenuAsync(AppListEntry entry)
+        {
+            var resultPinResult = PinResult.UnsupportedOs;
+            if (ApiInformation.IsTypePresent("Windows.UI.StartScreen.StartScreenManager"))
+            {
+                if (StartScreenManager.GetDefault().SupportsAppListEntry(entry))
+                {
+                    if (await StartScreenManager.GetDefault().ContainsAppListEntryAsync(entry))
+                    {
+                        resultPinResult = PinResult.PinAlreadyPresent;
+                    }
+                    else
+                    {
+                        if (await StartScreenManager.GetDefault().RequestAddAppListEntryAsync(entry))
+                        {
+                            resultPinResult = PinResult.PinPresent;
+                        }
+                        else
+                        {
+                            resultPinResult = PinResult.PinAlreadyPresent;
+                        }
+                    }
+                }
+                else
+                {
+                    resultPinResult = PinResult.UnsupportedDevice;
+                }
+            }
+
+            return resultPinResult;
+        }
+
+        /// <summary>
+        /// Pin Specific App in Windows StartMenu based on User
+        /// </summary>
+        /// <param name="user">User</param>
+        /// <param name="entry">AppListEntry</param>
+        /// <returns>PinResult</returns>
+        public static async Task<PinResult> PinUserSpecificAppToStartMenuAsync(User user, AppListEntry entry)
+        {
+            var resultPinResult = PinResult.UnsupportedOs;
+            if (ApiInformation.IsTypePresent("Windows.UI.StartScreen.StartScreenManager"))
+            {
+                if (StartScreenManager.GetForUser(user).SupportsAppListEntry(entry))
+                {
+                    if (await StartScreenManager.GetForUser(user).ContainsAppListEntryAsync(entry))
+                    {
+                        resultPinResult = PinResult.PinAlreadyPresent;
+                    }
+                    else
+                    {
+                        if (await StartScreenManager.GetForUser(user).RequestAddAppListEntryAsync(entry))
```

Comment:
returning false in method RequestAddAppListEntryAsync doesn't mean PinAlreadyPresent, it means the operation failed.
I see you need to add another enum that represents that the operation failed so the API consumer could retry if failed.
Please apply this comment on all similar methods.

Reply:
thanks , good catch

Reply:
I can't see the new changes my friend :)

Reply:
Can you please check now